### PR TITLE
SimplifyGlobals: Apply constant globals to segment offsets

### PR DESCRIPTION
**Pull request by kripken**
We already applied such globals to other globals, but can do the same to offsets
of data and element segments.

Suggested in https://github.com/WebAssembly/binaryen/issues/6220#issuecomment-1894983204

### CODE REVIEW

File: test/lit/passes/simplify-globals-offsets.wast [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      )
+    )
+    (drop
+      (table.get $table
+        (i32.const 0)
+      )
+    )
+    (drop
+      (global.get $use-defined)
+    )
```
Comment (aheejin) [inline test/lit/passes/simplify-globals-offsets.wast:75]: Maybe use `$use-imported` too as a comparison?

File: test/lit/passes/simplify-globals-offsets.wast [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  (global $use-defined i32 (global.get $defined))
+
+  ;; CHECK:      (data $use-imported (global.get $imported) "hello, world!")
+  (data $use-imported (global.get $imported) "hello, world!")
+
+  ;; CHECK:      (data $use-defined (i32.const 42) "hello, world!")
+  (data $use-defined (global.get $defined) "hello, world!")
+
+  ;; CHECK:      (data $dropped "hello, world!")
+  (data $dropped "hello, world!")
```
Comment (aheejin) [inline test/lit/passes/simplify-globals-offsets.wast:34]: Is this relevant to this PR?

**Review by aheejin — APPROVED** [review]

File: test/lit/passes/simplify-globals-offsets.wast (hunk 2 above)
Comment (kripken) [reply test/lit/passes/simplify-globals-offsets.wast:35]: It's a segment without an offset, so the pass has nothing to do there but also it should not error on it (there is a nullptr for the offset). I added a comment to clarify.

File: test/lit/passes/simplify-globals-offsets.wast (hunk 1 above)
Comment (kripken) [reply test/lit/passes/simplify-globals-offsets.wast:82]: Sounds good, done.
